File: DemoServer/AdditionalLanguages/python/autoIndexes/autoMapIndex2/autoMapIndex2.py
from demo_example import Example, RunParamsBase
from models import Employee

class RunParams(RunParamsBase):
    def __init__(self, country: str):
        self.country = country

class AutoMapIndex2(Example):
    def run(self, run_params: RunParams):
        country = run_params.country

        #region Demo
        with self.document_store_holder.store().open_session() as session:
            #region Step_1
            find_employee_query = (
                session.query(object_type=Employee)
                .where_equals("Address.Country", country)
                .where_starts_with("Title", "Sales")
            )
            #endregion

            #region Step_2
            employee_result = find_employee_query.first()
            #endregion
        #endregion

        return employee_result

File: DemoServer/AdditionalLanguages/php/autoIndexes/autoMapIndex2/AutoMapIndex2.php
<?php

namespace RavenDB\Demo\autoIndexes\autoMapIndex2;

//region Usings
//endregion

use RavenDB\Demo\common\DocumentStoreHolder;
use RavenDB\Demo\common\models\Employee;

class AutoMapIndex2
{
    public function __invoke(RunParams  $runParams): ?Employee
    {
        $country = $runParams->getCountry();

        //region Demo
        $employeeResult = null;

        $session = DocumentStoreHolder::getStore()->openSession();
        
        try {
            //region Step_1
            $findEmployeeQuery = $session->query(Employee::class)
                    ->whereEquals("Address.Country", $country)
                    ->whereStartsWith("Title", "Sales");
            //endregion

            //region Step_2
            $employeeResult = $findEmployeeQuery->firstOrDefault();
            //endregion
            
        } finally {
            $session->close();
        }
        //endregion

        return $employeeResult;
    }
}
